eppo_client/client.py

```diff
@@ -286,102 +288,112 @@ def get_bandit_action(
-            | subject_attributes.numeric_attributes,  # type: ignore
-            default,
-        )
-
-        # if the variation is not the bandit key, then the subject is not allocated in the bandit
-        if variation not in self.get_bandit_keys():
-            return BanditResult(variation, None)
-
-        # for now, assume that the variation is equal to the bandit key
-        bandit_data = self.__config_requestor.get_bandit_model(variation)
+    ) -> str | None:
+        # if no actions are given--a valid use case--return the variation with no action
+        if len(actions) == 0:
+            return None
+        
+        bandit_data = self.__config_requestor.get_bandit_model(bandit_key)
 
         if not bandit_data:
             logger.warning(
                 f"[Eppo SDK] No assigned action. Bandit not found for flag: {flag_key}"
             )
-            return BanditResult(variation, None)
+            return None
+        
+        subject_context_attributes = convert_subject_context_to_context_attributes(
+            subject_context
+        )
+        action_contexts = convert_actions_to_action_contexts(actions)
 
         evaluation = self.__bandit_evaluator.evaluate_bandit(
             flag_key,
             subject_key,
-            subject_attributes,
+            subject_context_attributes,
             action_contexts,
             bandit_data.bandit_model_data,
         )
 
         # log bandit action
-        bandit_event = {
-            "flagKey": flag_key,
-            "banditKey": bandit_data.bandit_key,
-            "subject": subject_key,
-            "action": evaluation.action_key if evaluation else None,
-            "actionProbability": evaluation.action_weight if evaluation else None,
-            "optimalityGap": evaluation.optimality_gap if evaluation else None,
-            "modelVersion": bandit_data.bandit_model_version if evaluation else None,
-            "timestamp": datetime.datetime.utcnow().isoformat(),
-            "subjectNumericAttributes": (
-                subject_attributes.numeric_attributes
-                if evaluation.subject_attributes
-                else {}
-            ),
-            "subjectCategoricalAttributes": (
-                subject_attributes.categorical_attributes
-                if evaluation.subject_attributes
-                else {}
-            ),
-            "actionNumericAttributes": (
-                evaluation.action_attributes.numeric_attributes
-                if evaluation.action_attributes
-                else {}
-            ),
-            "actionCategoricalAttributes": (
-                evaluation.action_attributes.categorical_attributes
-                if evaluation.action_attributes
-                else {}
-            ),
-            "metaData": {"sdkLanguage": "python", "sdkVersion": __version__},
-        }
-        self.__assignment_logger.log_bandit_action(bandit_event)
+        try:
```

Comment:
We didn't have a try-catch surrounding bandit logging, so putting this in place. This way, if the (customer-provided) bandit log method fails, the assignment is still recorded.